examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,133 +60,507 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
+
+      ts = PhaseEquil(e_int, qt_node, rho_node)
+      Phpart = PhasePartition(ts)
+      thermoQ[i,1,e] = Phpart.liq
+      thermoQ[i,2,e] = Phpart.ice
+      thermoQ[i,3,e] = qt_node-Phpart.liq-Phpart.ice
+      thermoQ[i,4,e] = ts.T
+      thermoQ[i,5,e] = liquid_ice_pottemp(ts)
+      thermoQ[i,6,e] = dry_pottemp(ts)
+      thermoQ[i,7,e] = virtual_pottemp(ts)
+    end
+  end
+
+  fluctT = zeros(Nq * Nq * Nqk, nthermo, nrealelem)    
+  fluctQ = zeros(Nq * Nq * Nqk, nstate, nrealelem)
+  varQ   = zeros(Nq * Nq * Nqk, nstate, nrealelem)
+
+  rho_localtot = sum(localQ[:, 1, :]) / (size(localQ, 1) * size(localQ, 3))
+  U_localtot   = sum(localQ[:, 2, :])
+  V_localtot   = sum(localQ[:, 3, :])
+  W_localtot   = sum(localQ[:, 4, :]) / (size(localQ, 1) * size(localQ, 3))
+  e_localtot   = sum(localQ[:, 5, :])
+  qt_localtot  = sum(localQ[:, 6, :])
+
+  qliq_localtot  = sum(thermoQ[:, 1, :])
+  qice_localtot  = sum(thermoQ[:, 2, :])
+  qvap_localtot  = sum(thermoQ[:, 3, :])
+  T_localtot     = sum(thermoQ[:, 4, :])
+  theta_localtot = sum(thermoQ[:, 5, :])
+
+  rho_tot   = MPI.Reduce(rho_localtot, +, 0, MPI.COMM_WORLD)
+  U_tot     = MPI.Reduce(U_localtot, +, 0, MPI.COMM_WORLD)
+  V_tot     = MPI.Reduce(V_localtot, +, 0, MPI.COMM_WORLD)
+  W_tot     = MPI.Reduce(W_localtot, +, 0, MPI.COMM_WORLD)
+  e_tot     = MPI.Reduce(e_localtot, +, 0, MPI.COMM_WORLD)
+  qt_tot    = MPI.Reduce(qt_localtot, +, 0, MPI.COMM_WORLD)
+  qliq_tot  = MPI.Reduce(qliq_localtot, +, 0, MPI.COMM_WORLD)
+  qice_tot  = MPI.Reduce(qice_localtot, +, 0, MPI.COMM_WORLD)
+  qvap_tot  = MPI.Reduce(qvap_localtot, +, 0, MPI.COMM_WORLD)
+  T_tot     = MPI.Reduce(T_localtot, +, 0, MPI.COMM_WORLD)
+  theta_tot = MPI.Reduce(theta_localtot, +, 0, MPI.COMM_WORLD)
+
+  if mpirank == 0
+    rho_avg   = rho_tot / (nranks)
+    U_avg     = (U_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    V_avg     = (V_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    W_avg     = (W_tot / (nranks * rho_avg)) 
+    e_avg     = (e_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qt_avg    = (qt_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qliq_avg  = (qliq_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qice_avg  = (qice_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qvap_avg  = (qvap_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    T_avg     = (T_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    theta_avg = (theta_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+
+    @debug "ρ average = $(rho_avg)"
+    @debug "U average = $(U_avg)"
+    @debug "V average = $(V_avg)"
+    @debug "W average = $(W_avg)"
+    @debug "e average = $(e_avg)"
+    @debug "qt average = $(qt_avg)"
+    @debug "qliq average = $(qliq_avg)"
+    @debug "qice average = $(qice_avg)"
+    @debug "qvap average = $(qvap_avg)"
+    @debug "T average = $(T_avg)"
+    @debug "theta average = $(theta_avg)"
+  end
```

Comment:
This should be removed in full or corrected. Not all ranks have the same number of points.